moveit_planners/chomp/chomp_interface/src/chomp_plugin.cpp

```diff
@@ -32,34 +32,44 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  *********************************************************************/
 
-#include <memory>
-
+#include <chomp_interface/chomp_planning_context.h>
+#include <moveit/collision_distance_field/collision_detector_allocator_hybrid.h>
 #include <moveit/planning_interface/planning_interface.h>
 #include <moveit/planning_scene/planning_scene.h>
 #include <moveit/robot_model/robot_model.h>
-#include <moveit/collision_distance_field/collision_detector_allocator_hybrid.h>
-#include <chomp_interface/chomp_planning_context.h>
 
 #include <pluginlib/class_list_macros.hpp>
+#include <vector>
 
 namespace chomp_interface
 {
+static const rclcpp::Logger LOGGER = rclcpp::get_logger("chomp_optimizer");
+
 class CHOMPPlannerManager : public planning_interface::PlannerManager
 {
 public:
   CHOMPPlannerManager() : planning_interface::PlannerManager()
   {
   }
 
-  bool initialize(const moveit::core::RobotModelConstPtr& model, const std::string& ns) override
+  bool initialize(const moveit::core::RobotModelConstPtr& model, const rclcpp::Node::SharedPtr& node,
+                  const std::string& parameter_namespace) override
   {
-    ros::NodeHandle nh("~");
-    if (!ns.empty())
-      nh = ros::NodeHandle(ns);
+    std::string actual_ns;
+    if (parameter_namespace.empty())
+    {
+      actual_ns = "~";
+    }
+    else
+    {
+      actual_ns = parameter_namespace;
+    }
+    std::shared_ptr<rclcpp::Node> nh_ptr = std::make_shared<rclcpp::Node>(actual_ns);
```

Comment:
Why not use the input node ? also did you test having `~` as a namespace ? I think it will throw an exception

Reply:
This was closest to the original code. I removed the `~` namespace part, and it's now using the input node.